### PR TITLE
Refactor main.js: down to 450 lines

### PR DESCRIPTION
**Pull request by dcposch**
* Move all the torrent event handlers to their own file `torrent-controller.js`
* Move all the file extension checking to the same place

### CODE REVIEW

**Comment by dcposch** [comment]
@feross @mathiasvr

File: renderer/controllers/torrent-controller.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  // Gets a torrent summary {name, infoHash, status} from state.saved.torrents
+  // Returns undefined if we don't know that infoHash
+  getTorrentSummary (torrentKey) {
+    return TorrentSummary.getByKey(this.state, torrentKey)
+  }
+}
+
+function getTorrentPath (torrentSummary) {
+  var itemPath = path.join(torrentSummary.path, torrentSummary.files[0].path)
```
Comment (mathiasvr) [inline renderer/controllers/torrent-controller.js]: We probably want to use TorrentSummary.getFileOrFolder here.

**Comment by mathiasvr** [comment]
LGTM 👍

File: renderer/controllers/torrent-controller.js (hunk 1 above)
Comment (dcposch) [inline renderer/controllers/torrent-controller.js]: good call, done

**Pull request merged**